### PR TITLE
Download tab (await data at the same time)

### PR DESCRIPTION
**Pull request by dippindots**
waiting for data at the same time

Changes proposed in this pull request:
- a: download tab (await data at the same time)

From:
image

To:
image

### CODE REVIEW

**Comment by alisman** [comment]
@dippindots is this connect with an issue?

**Comment by dippindots** [comment]
@alisman No, there is no ticket described it. But you can try these two links to see the difference.
[link(before)](http://www.cbioportal.org/results/download?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_list=msk_impact_2017&case_set_id=msk_impact_2017_cnaseq&clinicallist=NUM_SAMPLES_PER_PATIENT&data_priority=0&gene_list=EGFR&geneset_list=%20&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=msk_impact_2017_cna&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=msk_impact_2017_mutations&show_samples=false&tab_index=tab_visualize), [link(later)](https://cbioportal-frontend-pr-2660.herokuapp.com/results/download?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_list=msk_impact_2017&case_set_id=msk_impact_2017_cnaseq&clinicallist=NUM_SAMPLES_PER_PATIENT&data_priority=0&gene_list=EGFR&geneset_list=%20&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=msk_impact_2017_cna&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=msk_impact_2017_mutations&show_samples=false&tab_index=tab_visualize)

File: src/pages/resultsView/download/DownloadTab.tsx [hunk 1]
```diff
@@ -398,9 +399,9 @@ export default class DownloadTab extends React.Component<IDownloadTabProps, {}>
                     {hasValidMutationData(this.mutationData.result!) && this.mutationDownloadControls()}
                     {hasValidData(this.mrnaData.result!) && this.mrnaExprDownloadControls()}
                     {hasValidData(this.proteinData.result!) && this.proteinExprDownloadControls()}
-                    {this.alteredSamplesDownloadControls.component}
-                    {this.unalteredSamplesDownloadControls.component}
-                    {this.sampleMatrixDownloadControls.component}
+                    {this.alteredSamplesDownloadControls(this.alteredCaseAlterationData.result!, this.props.store.virtualStudyParams.result!)}
```
Comment (alisman) [inline src/pages/resultsView/download/DownloadTab.tsx]: lets take this jsx out of the remoteData.  i've never believed that rendering should be done in return of invoke function.    the remoteData should just return data and that should be consumed by component render method.

Comment (dippindots) [reply src/pages/resultsView/download/DownloadTab.tsx]: @alisman Sure, so we remove this `handleCloseModal` and move the table in the render function, is that alright?

Comment (alisman) [reply src/pages/resultsView/download/DownloadTab.tsx]: correct.  MobxView is permissable, but I actually still don't really like it.   so for this, lets just put  the code in render method.

Comment (dippindots) [reply src/pages/resultsView/download/DownloadTab.tsx]: @alisman Updated, is it looks better?

**Pull request merged**